examples/evaluation/evaluate_with_custom_code_metrics.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import os
+
+import openai
+import pandas as pd
+
+import mlflow
+from mlflow.metrics import make_metric
+from mlflow.metrics.base import MetricValue
+from mlflow.metrics.metric_definitions import standard_aggregations
+
+logging.getLogger("mlflow").setLevel(logging.ERROR)
+
+
+# Uncomment the following lines to run this script without using a real OpenAI API key.
+# os.environ["OPENAI_API_KEY"] = "test"
+
+assert "OPENAI_API_KEY" in os.environ, "Please set the OPENAI_API_KEY environment variable."
+
+
+# Helper function to check if a string is valid python code
+def is_valid_python_code(code: str) -> bool:
+    try:
+        compile(code, "<string>", "exec")
+        return True
+    except SyntaxError:
+        return False
+
+
+# Create an evaluation function that iterates through the predictions
+def eval_fn(eval_df, metrics):
+    y_pred = eval_df["prediction"]
```

Comment:
With the new code. we can change the code to be:

def eval_fn(prediction):
   y_pred = prediction

Reply:
this change does not work 
```
(mlflow-dev-env) (base) prithvi.kannan@NR93P4PJF3 mlflow % python examples/evaluation/evaluate_with_custom_code_metrics.py
Using default facebook/roberta-hate-speech-dynabench-r4-target checkpoint
Using pad_token, but it is not set yet.
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 17.32it/s]
Traceback (most recent call last):
  File "/Users/prithvi.kannan/mlflow/examples/evaluation/evaluate_with_custom_code_metrics.py", line 67, in <module>
    results = mlflow.evaluate(
  File "/Users/prithvi.kannan/mlflow/mlflow/models/evaluation/base.py", line 1629, in evaluate
    evaluate_result = _evaluate(
  File "/Users/prithvi.kannan/mlflow/mlflow/models/evaluation/base.py", line 1060, in _evaluate
    eval_result = evaluator.evaluate(
  File "/Users/prithvi.kannan/mlflow/mlflow/models/evaluation/default_evaluator.py", line 1591, in evaluate
    evaluation_result = self._evaluate(model, is_baseline_model=False)
  File "/Users/prithvi.kannan/mlflow/mlflow/models/evaluation/default_evaluator.py", line 1508, in _evaluate
    self._evaluate_extra_metrics(eval_df)
  File "/Users/prithvi.kannan/mlflow/mlflow/models/evaluation/default_evaluator.py", line 1153, in _evaluate_extra_metrics
    eval_fn_args = self._get_args_for_metrics(extra_metric, eval_df)
  File "/Users/prithvi.kannan/mlflow/mlflow/models/evaluation/default_evaluator.py", line 1132, in _get_args_for_metrics
    column = self.col_mapping.get(param_name, param_name)
AttributeError: 'NoneType' object has no attribute 'get'
```

do we need to fix this somewhere else?

Reply:
Sorry its `predictions`: https://github.com/mlflow/mlflow/blob/mlflow-evaluation/mlflow/models/evaluation/base.py#L90

Can we try that?

Reply:
works, updated!